### PR TITLE
remove screenshot in browser observation

### LINKED ISSUE #1553: [Refactor]: Better screenshot rremoval
**Summary**

We have screenshot removal in several different places in the codebase, and we should probably consolidate the code into a single place.
More discussion: https://github.com/OpenDevin/OpenDevin/pull/1550

### PR DESCRIPTION
**Pull request by frankxu2004**
Put all the removal of screenshot logic inside to_memory for browser obs.

Fixes #1553

### CODE REVIEW

File: opendevin/events/observation/browse.py [hunk 1]
```diff
@@ -5,35 +5,65 @@
 from .observation import Observation
 
 
+def _remove_fields(obj, fields: set[str]):
```
Comment (rbren) [inline opendevin/events/observation/browse.py:8]: might make sense to put `_remove_fields` in the `events` folder so other actions/observations can use it

File: opendevin/events/observation/browse.py [hunk 2]
```diff
@@ -5,35 +5,65 @@
...
+            _remove_fields(value, fields)
+    elif isinstance(obj, list) or isinstance(obj, tuple):
+        for item in obj:
+            _remove_fields(item, fields)
+    elif hasattr(obj, '__dataclass_fields__'):
+        for field in fields:
+            if field in obj.__dataclass_fields__:
+                setattr(obj, field, None)
+        for value in obj.__dict__.values():
+            _remove_fields(value, fields)
```
Comment (rbren) [inline opendevin/events/observation/browse.py:30]: do we need this part? Seems like the first `if` covers everything

**Review by rbren — APPROVED** [review]

File: opendevin/events/observation/browse.py (hunk 2 above)
Comment (frankxu2004) [reply opendevin/events/observation/browse.py]: Now that I moved this util function to under `events`, I think we could keep this function work for not only `dict`, but also `dataclass` or `list` of `dict`, etc.

**Comment by frankxu2004** [comment]
Just updated!

File: opendevin/events/utils.py [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+                del obj[field]
+        for _, value in obj.items():
+            _remove_fields(value, fields)
+    elif isinstance(obj, list) or isinstance(obj, tuple):
+        for item in obj:
+            _remove_fields(item, fields)
+    elif hasattr(obj, '__dataclass_fields__'):
+        for field in fields:
+            if field in obj.__dataclass_fields__:
+                setattr(obj, field, None)
```
Comment (enyst) [inline opendevin/events/utils.py:21]: Will all fields removed this way accept None? It seems BrowserOutputObservation screenshot doesn't and we initialize it somewhere with ''

Comment (frankxu2004) [reply opendevin/events/utils.py]: fixed this by removing support for other classes. Now this function will only work with dictionaries, as we only use it with `to_memory()` and all the observation textual representation used for agents should come from `observation.to_memory()`.

File: opendevin/events/observation/browse.py (hunk 2 above)
Comment (frankxu2004) [reply opendevin/events/observation/browse.py]: actually removed this part.

**Comment by rbren** [comment]
Looks like a couple integration tests are unhappy for some reason...

**Comment by rbren** [comment]
Thanks Frank!

**Pull request merged**